addons/osfstorage/models.py

```diff
@@ -591,3 +595,20 @@ def create_waterbutler_log(self, auth, action, metadata):
             auth=auth,
             params=params
         )
+
+    @property
+    def storage_usage(self):
+        storage_usage = cache.get('storage_usage:' + self._id)
+
+        if storage_usage:
+            return storage_usage
+
+        file_versions = OsfStorageFile.objects.filter(provider='osfstorage',
+                                                   target_object_id=self.owner_id,
+                                                   versions__isnull=False).values_list('versions',
+                                                                                       flat=True)
+        # An empty queryset will produce a None instead of 0
+        storage_usage_total = FileVersion.objects.filter(id__in=file_versions).aggregate(sum=models.Sum('size'))['sum'] or 0
```

Comment:
For performance it might be much faster to write this as a single query...

```python
AbstractNode.objects.filter(
    files__type='osf.osfstoragefile',
    id=self.owner_id
).annotate(
    Sum('files__versions__size'),
).values_list(
    'files__versions__size__sum',
).first()
```

Reply:
http://tatiyants.com/pev/#/plans/plan_1539805418000